src/test/groovy/org/gradle/android/SimpleAndroidApp.groovy

```diff
@@ -289,10 +289,16 @@ class SimpleAndroidApp {
     }
 
     private String getSourceCompatibilityIfEnabled() {
-        return (sourceCompatibility != null) ? """
+        def currentSourceCompatibility = sourceCompatibility
```

Comment:
An alternative is to always set this sourceCompatbility. I chose not to do this since I think it adds build time costs to run the transformations. That being said since Kotlin is applied in almost all tests it is probably always applied anyways.